armtesting/arm-none-eabi.md

```diff
@@ -0,0 +1,293 @@
+a suitable linker script.
+
+[`cortex-m-rt`]: https://github.com/rust-embedded/cortex-m-rt
+
+Targets named `thumb*` instead of `arm*`
+generate Thumb-mode code by default. M-profile processors (`thumbv*m*-*`
+targets) only support Thumb-mode code.
+For the `arm*` targets, Thumb-mode code generation can be enabled by using
+`-C target-feature=+thumb-mode`. Using the unstable
+`#![feature(arm_target_feature)]`, the attribute
+`#[target_feature(enable = "thumb-mode")]` can be applied to individual
+`unsafe` functions to cause those functions to be compiled to Thumb-mode code.
+
+## Building Rust Programs
+
+For the Tier 3 targets in this family, Rust does not ship pre-compiled
+artifacts.
+
+Use the `build-std` nightly cargo feature to build the `core` library. You
+can pass this as a command line argument to cargo, or your `.cargo/config.toml`
+file might include the following lines:
+
+```toml
+[unstable]
+build-std = ["core"]
+```
+
+Most of `core` should work as expected, with the following notes:
+* If the target is not `*hf`, then floating-point operations are emulated in
+  software.
+* Integer division is also emulated in software on some targets, depending on
+  the CPU.
+* Architectures prior to ARMv7 don't have atomic instructions.
+
+`alloc` is also supported, as long as you provide your own global allocator.
+
+Rust programs are output as ELF files.
+
+## Testing
+
+This is a cross-compiled target that you will need to emulate during testing.
+
+The exact emulator that you'll need depends on the specific device you want to
+run your code on.
+
+## Cross-compilation toolchains and C code
+
+The target supports C code compiled with the `arm-none-eabi` target triple and
+a suitable `-march` or `-mcpu` flag.
+
+`gcc` or `clang` can be used, but note that `gcc` uses `-fshort-enums` by
+default for `arm-none*` targets, while `clang` does not. `rustc` matches the
+`gcc` behavior, i.e., the size of a `#[repr(C)] enum` in Rust can be as little
+as 1 byte, rather than 4, as they are on `arm-linux` targets.
+
+## Floating point on Cortex-M processors
+
+
+The `*hf` targets assume that the device has a hardware FPU and lowers all
+floating point operations to hardware instructions.  Additionally, this target
+uses the "hard" float ABI, where floating point values are passed to/from
+subroutines via FPU registers.
+
+To opt in to double precision hardware FPU support on Cortex-M7, use the
+`-C target-cpu=cortex-m7` flags.
+
+It is also possible to use the hardware FPU while using the soft float ABI, by
+using the `thumbv7em-none-eabi` target and passing `-C target-cpu=cortex-m4`
+(or `cortex-m7`). In this case the FPU must be enabled at startup, before
+entering any function which takes floating-point arguments or uses any
+floating-point variables. The [`cortex-m`] crate provides the
+`SCB::enable_fpu()` convenience function to enable the FPU.
+
+## Target flags
+
+CPU-specific code generation and features can be enabled using the
```

Comment:
I should note here that the feature flags are unstable and may change. I think the target CPUs are stable? Or at least rustc doesn't complain about them.